cpp/tests/utilities/column_utilities.cu

```diff
@@ -213,30 +457,43 @@ template <typename T, bool check_exact_equality>
 struct column_comparator_impl {
   void operator()(column_view const& lhs,
                   column_view const& rhs,
+                  column_view const& lhs_row_indices,
+                  column_view const& rhs_row_indices,
                   bool print_all_differences,
+                  bool throw_on_fail,
                   int depth)
   {
     auto d_lhs = cudf::table_device_view::create(table_view{{lhs}});
     auto d_rhs = cudf::table_device_view::create(table_view{{rhs}});
 
+    auto d_lhs_row_indices = cudf::column_device_view::create(lhs_row_indices);
+    auto d_rhs_row_indices = cudf::column_device_view::create(rhs_row_indices);
+
     using ComparatorType = std::conditional_t<check_exact_equality,
                                               corresponding_rows_unequal,
                                               corresponding_rows_not_equivalent>;
 
     auto differences = rmm::device_uvector<int>(
       lhs.size(), rmm::cuda_stream_default);  // worst case: everything different
-    auto diff_iter = thrust::copy_if(rmm::exec_policy(),
-                                     thrust::make_counting_iterator(0),
-                                     thrust::make_counting_iterator(lhs.size()),
-                                     differences.begin(),
-                                     ComparatorType(*d_lhs, *d_rhs));
+    auto input_iter = thrust::make_counting_iterator(0);
+    auto diff_iter =
+      thrust::copy_if(rmm::exec_policy(),
+                      input_iter,
+                      input_iter + lhs_row_indices.size(),
+                      differences.begin(),
+                      ComparatorType(*d_lhs, *d_rhs, *d_lhs_row_indices, *d_rhs_row_indices));
 
     differences.resize(thrust::distance(differences.begin(), diff_iter),
                        rmm::cuda_stream_default);  // shrink back down
 
-    if (not differences.is_empty())
-      GTEST_FAIL() << stringify_column_differences(
-        differences, lhs, rhs, print_all_differences, depth);
+    if (not differences.is_empty()) {
+      if (throw_on_fail) {
+        CUDF_FAIL("Column differences detected");
```

Comment:
Would it be difficult to add differences just like in the print case to the exception message?

Reply:
We could, but the intent of the feature was to allow stuff like this to pass quietly without spewing output:

`EXPECT_THROW(cudf::test::expect_columns_equal(*c0, *c1, false, true), cudf::logic_error);`